patroni/ha.py

```diff
@@ -1113,8 +1113,8 @@ def _run_cycle(self):
                         self.dcs.delete_leader()
                         self.dcs.reset_cluster()
                         return 'removed leader lock because postgres is not running'
-                    elif not (self.state_handler.rewind_executed or
-                              self.state_handler.need_rewind and self.state_handler.can_rewind):
+                    elif (self.state_handler.rewind_executed or
```

Comment:
It feels wrong to me.
We want to start postgres if pg_rewind was executed successfully and don't want to start it if pg_rewind failed.
The new version will not try to start postgres even after successful pg_rewind.